rotkehlchen/chain/evm/contracts.py

```diff
@@ -215,17 +328,18 @@ def abi_or_none(self, name: str) -> Optional[list[dict[str, Any]]]:
         with GlobalDBHandler().conn.read_ctx() as cursor:
             cursor.execute('SELECT value FROM contract_abi WHERE name=?', (name,))
             result = cursor.fetchone()
-            if result is None:
-                return None
-            try:
-                abi_data = json.loads(result[0])
-            except json.decoder.JSONDecodeError as e:
-                log.error(
-                    f'Failed to decode {name} abi {result[0]} from DB as json due to {str(e)}',
-                )
-                return None
-
-            return abi_data
+
+        if result is None:
+            return try_download_abi_from_remote(name)
+        try:
+            abi_data = json.loads(result[0])
+        except json.decoder.JSONDecodeError as e:
+            log.error(
+                f'Failed to decode {name} abi {result[0]} from DB as json due to {str(e)}',
+            )
+            return None
+
+        return abi_data
```

Comment:
this changes is just because the cursor  can be closed earlier right?

Reply:
Yes

Reply:
Yes